src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,103 @@
+package duke.parser;
+
+import java.util.List;
+import duke.task.*;
+import duke.command.*;
+import duke.exception.*;
+import duke.ui.*;
+
+public class Parser {
+    public static Command<List<Task>> parseInput(String userInput) throws DukeException {
+        try {
+            String[] parts = userInput.split(" ", 2);
+            String command = parts[0].toLowerCase();
+            List<String> arguments = parts.length > 1 ? List.of(parts[1].split("\\s+")) : List.of();
+
+            String taskDescription = extractTaskDescription(arguments);
+            String deadline = "Error: No deadline set.";
+            String start = "Error: No start time set.";
+            String end = "Error: No end time set.";
+
+            // Get Deadline
+            if (arguments.contains("/by")) {
+                int index = arguments.indexOf("/by");
+                if (index < arguments.size() - 1) {
+                    deadline = String.join(" ", arguments.subList(index + 1, arguments.size()));
+                } else {
+                    throw new DukeException("Error: Missing deadline after /by. Correct format: /by <deadline>");
+                }
+            } else if (command.equals("deadline")) {
+                throw new DukeException("Error: No description provided for the deadline command.\nCorrect format: deadline <description> /by <deadline>\n");
+            }
+
+            // Get From / To values
+            if (arguments.contains("/from") && arguments.contains("/to")) {
+                int fromIndex = arguments.indexOf("/from");
+                int toIndex = arguments.indexOf("/to");
+                if (fromIndex < toIndex && toIndex < arguments.size() - 1) {
+                    start = String.join(" ", arguments.subList(fromIndex + 1, toIndex));
+                    end = String.join(" ", arguments.subList(toIndex + 1, arguments.size()));
+                } else {
+                    throw new DukeException("Error: Missing /from or /to after event command. Correct format: event <description> /from <start> /to <end>");
+                }
+            } else if (arguments.contains("/from") || arguments.contains("/to")) {
+                throw new DukeException("Error: Both /from and /to are required for the event command. Correct format: /from <start> /to <end>");
+            }
+
+            switch (command) {
+                case "help":
```

Comment:
No indentation for `case`  clauses in the `switch` , Can we fix this to adhere with the coding standard?